[Post 4.10.1] Add sample for activity grouping (#3415)
microsoft/BotFramework-WebChat: samples/02.branding-styling-and-customization/j.activity-grouping/README.md

- For both side
   - Bubble background will be `#F4F4F4`
   - Bubble border will be 2 pixels wide of color `#F4F4F4`
   - Bubble radius will be 4 pixels
   - Bubble nub will be 10 pixels in size and aligned to top
- Timestamps will be grouped if they are within 3 seconds
- Avatar will be shown in every timestamp group

```diff
  const res = await fetch('https://webchat-mockbot.azurewebsites.net/directline/token', { method: 'POST' });
  const { token } = await res.json();

  renderWebChat(
    {
+     className: 'webchat__chat',
-     directLine: createDirectLine({ token })
+     directLine: createDirectLine({ token }),
+     styleOptions: {
+       botAvatarInitials: 'Bot',
+       userAvatarInitials: 'You',
+
+       bubbleBackground: '#F4F4F4',
+       bubbleBorderColor: '#F4F4F4',
+       bubbleBorderRadius: 4,
+       bubbleBorderWidth: 2,
+       bubbleNubOffset: 0,
+       bubbleNubSize: 10,
+
+       bubbleFromUserBackground: '#F4F4F4',
+       bubbleFromUserBorderColor: '#F4F4F4',
+       bubbleFromUserBorderRadius: 4,
+       bubbleFromUserBorderWidth: 2,
+       bubbleFromUserNubOffset: 0,
+       bubbleFromUserNubSize: 10,
+
+       groupTimestamp: 3000,

> +       showAvatarInGroup: 'timestamp'

Author: Added:

- Timestamps will be grouped if they are within 3 seconds
   - By default, timestamps are always grouped regardless of their intervals
- Avatar will be shown in every timestamp/status group, this is default
   - Set it to `"sender"` to show avatar in every sender group
   - Set it to `true` to show avatar on every activity

# Sample - Activity grouping

This sample shows how to customize the activity grouping behavior.

![Sample UI](ui-1.png)

# Test out the hosted sample

-  [Try out MockBot](https://microsoft.github.io/BotFramework-WebChat/02.branding-styling-and-customization/j.activity-grouping)

# How to run

-  Fork this repository
-  Navigate to `/Your-Local-WebChat/samples/02.branding-styling-and-customization/j.activity-grouping` in command line
-  Run `npx serve`
-  Browse to [http://localhost:5000/](http://localhost:5000/)

# Things to try out

1. Type `echo Hello, World!` to the bot
1. Type `proactive` to the bot

> 1. Resizing the browser window

Reviewer: ```suggestion
1. Resize the browser window
```

# Code

> Jump to [completed code](#completed-code) to see the end-result `index.html`.

## Getting started

> Note: this sample is based from [`01.getting-started/a.full-bundle`](https://github.com/microsoft/BotFramework-WebChat/tree/master/samples/01.getting-started/a.full-bundle).

This sample will customize the activity grouping behavior, avatar UI, and bubble UI.

### Styling the container

```css
html,
body {
   height: 100%;
}

body {
   background-color: #fafafa;
   margin: 0;
}

#webchat {
   height: 100%;
   margin: 0;
   transition-duration: 0.5s;
   transition-property: margin;
}

.webchat__chat {
   background-color: white;
   box-shadow: 0 0 10px rgba(0, 0, 0, 0.05);
   display: flex;
   flex-direction: column;
   height: 100%;
   margin: auto;
   max-width: 360px;
   min-width: 360px;
   transition-duration: 0.5s;
   transition-property: max-width, min-width, width;
}

@media (min-width: 720px) {
   #webchat {
      margin: 0 10px;
   }

   .webchat__chat {
      max-width: 480px;
      min-width: 480px;
      width: 480px;
   }
}
```

### Setting `styleOptions`

The style options in the sample are set to the following:

> -  Bot side avatar initial is set to "Bot"

Reviewer: ```suggestion
-  Bot side avatar initials are set to "Bot"
```

> -  User side avatar initial is set to "You"

Reviewer: ```suggestion
-  User side avatar initials are set to "You"
```

> -  For both side

Reviewer: ```suggestion
-  For both sides:
```

>    -  Bubble background will be `#F4F4F4`

Reviewer: ```suggestion
   -  Bubble background is `#F4F4F4`
```

>    -  Bubble border will be 2 pixels wide of color `#F4F4F4`

Reviewer: ```suggestion
   -  Bubble border is 2 pixels wide of color `#F4F4F4`
```

>    -  Bubble radius will be 4 pixels

Reviewer: ```suggestion
   -  Bubble radius is 4 pixels
```

>    -  Bubble nub will be 10 pixels in size and aligned to top

Reviewer: ```suggestion
   -  Bubble nub is 10 pixels in size and aligned to top
```

> -  Timestamps will be grouped if they are within 3 seconds

Reviewer: ```suggestion
-  Timestamps within 3 seconds are grouped
```

>    -  By default, timestamps are always grouped regardless of their intervals

Reviewer: Is this line necessary?

Author: Chris said he want to know the default.

Reviewer: 👍🏼

> -  Avatar will be shown in every timestamp/status group, this is default

Reviewer: ```suggestion
-  The avatar is shown in every timestamp/status group, by default
```

>    -  Set it to `"sender"` to show avatar in every sender group

Reviewer: ```suggestion
   -  `"sender"` shows the avatar in every sender group
```

>    -  Set it to `true` to show avatar on every activity

Reviewer: ```suggestion
   -  `true` shows the avatar on every activity
```
